File: Java/problem1505-1.java
class Solution {
    public String minInteger(String num, int k) {
        if (k == 0 || num == null) {
            return num;
        }
        for (int i = 0; i <= 9; ++i) {
            int idx = num.indexOf(String.valueOf(i));
            if (idx >= 0 && idx <= k) {
                return String.valueOf(i) + minInteger(num.substring(0, idx) + num.substring(idx + 1, num.length()), k - idx);
            }
        }
        return num;
    }
}

File: python/problem1505-1.py
class Solution:
    
    def minInteger(self, num: str, k: int) -> str:
        if k == 0 or not num:
            return num
        for i in range(10):
            idx = num.find(str(i))
            if 0 <= idx <= k:
                return num[idx] + self.minInteger(num[0:idx] + num[idx + 1:], k - idx)
